src/main/proxy-functions/kube-api-request.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { chunk } from "lodash";
+import net from "net";
+import url from "url";
+import { apiKubePrefix } from "../../common/vars";
+import { ClusterManager } from "../cluster-manager";
+import type { ProxyApiRequestArgs } from "./types";
+
+const skipRawHeaders = new Set(["Host", "Authorization"]);
```

Comment:
could just be an array and use `includes` below instead of `Set.has`? Maybe there's an advantage to using `Set` here?

Reply:
I generally figure that a hashset has complexity of a hash and then a comparison.

Whereas the array would have a complexity of the length of the string being compared by the length of the array.